File: internship_backend_apr25/jury_solutions/python/b.py
def main():
    _ = input()  # Игнорируем первую строку ввода
    houses = list(map(int, input().split()))
    heaters = list(map(int, input().split()))

    min_radius = [float('inf')] * len(houses)

    j = 0
    k = 0

    houses.sort()
    heaters.sort()

    while j < len(houses):
        min_radius[j] = min(min_radius[j], abs(houses[j] - heaters[k]))

        if k > 0:
            min_radius[j] = min(min_radius[j], abs(houses[j] - heaters[k - 1]))

        if heaters[k] < houses[j] and k < len(heaters) - 1:
            k += 1
        else:
            j += 1

    print(max(min_radius))

if __name__ == "__main__":
    main()

File: internship_backend_apr25/jury_solutions/java/B.java
import java.util.Arrays;
import java.util.Scanner;

public class B {
	public static void main(String[] args) throws Exception {
		new B().run();
	}

	public void run() {
		Scanner scanner = new Scanner(System.in);
		scanner.nextLine();
		int[] houses = Arrays.stream(scanner.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray();
		int[] heaters = Arrays.stream(scanner.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray();

		int[] minRadius = new int[houses.length];
		Arrays.fill(minRadius, Integer.MAX_VALUE);

		int j = 0;
		int k = 0;

		Arrays.sort(houses);
		Arrays.sort(heaters);

		while (j < houses.length) {
			minRadius[j] = Math.min(minRadius[j], Math.abs(houses[j] - heaters[k]));

			if (k > 0)
				minRadius[j] = Math.min(minRadius[j], Math.abs(houses[j] - heaters[k - 1]));

			if (heaters[k] < houses[j] && k < heaters.length - 1)
				k++;
			else
				j++;
		}

		System.out.println(Arrays.stream(minRadius).max().getAsInt());
	}
}

